### PR TITLE
PR-715 => develop

### PR DESCRIPTION
**Pull request by nmsger-isento**
# Related PRs:
- https://github.com/pib-rocks/cerebra/pull/58 

# Jira Ticket:

Link to corresponding jira ticket:
https://pib-rocks.atlassian.net/jira/software/c/projects/PR/boards/3?selectedIssue=PR-715

# Checklist:
These tasks need to be completed before the pull request can be merged:
- [ ] The code has been reviewed and approved at least once
- [x] All necessary tests have been successful
- [ ] The product owner reviewed and accepted the story

# Testing:
### Test acceptance criteria:

- [x] The local installation via setup scripts works as before
- [x] The docker installation via setup scripts works, docker containers start when booting pib (use `sudo docker ps`)
- [x] Installing the docker installation on Ubuntu 24.02 via setup script will also apply the correct system settings (the containers themselves won't work yet due to an issue with Tinkerforge's brick daemon)

When testing, consider the additional changes the setup script usually applies, such as:
- [x] The setup script produces a log output that can be found in the $HOME directory
- [x] Default programs that should get deleted during the setup process get deleted
- [x] System settings get changed (Power settings, display resolution, pinning Terminal to favorites, etc.)
- [x] pib eyes are moved to the desktop
- [x] Cerebra runs on port 80
- [x] Connecting to pib via SSH works
- [x] Directory structure after using the local install still is the same

### Setup Tests:

Install Cerebra via setup script using the local installation method: `bash setup-pib.sh -b=PR-715 -f=PR-715 -l`
Install Cerebra via setup script using the docker installation method: `bash setup-pib.sh -b=PR-715 -f=PR-715`
A setup test was successfull on the following system(s):
- [x] Setup on a physical pib
- [x] Setup on a virtual machine
- [x] Setup with docker

### CODE REVIEW

File: setup/installation_scripts/local_install.sh [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+  git clone --recurse-submodules https://github.com/luxonis/depthai-python.git
+  cd depthai-python/examples || { print ERROR "depthai-python/examples not found"; return 1; }
+  python3 install_requirements.py
+  # Hand tracker
+  git clone https://github.com/geaxgx/depthai_hand_tracker.git
+  cd depthai_hand_tracker || { print ERROR "depthai_hand_tracker not found"; return 1; }
+  pip install -r requirements.txt
+  cd "$HOME" || { print ERROR "${HOME} not found"; return 1; }
+
+  # SLAM dependencies (optional)
```
[the remark this answers is not in the document]
Comment (nmsger-isento) [reply setup/installation_scripts/local_install.sh:146]: SLAM dependencies were introduced in an earlier PR and should be installed together with pib. Given that we currently don't use them (and they don't run on a Raspy 4) I think keeping them in the setup script like this allows us to quickly reuse them in the future. An alternative would be to install them again, however, they are 1-2 GB large if I remember it correctly.

File: setup/setup-pib.sh [hunk 2]
```diff
@@ -1,222 +1,262 @@
...
-		installed_packages_to_be_removed+="$package_name "
-	fi
+# VALIDATE CLI ARGUMENTS
+BRANCH_BACKEND="main"
+BRANCH_FRONTEND="main"
+INSTALL_METHOD="docker"
+# Check if branch was specified
+while [ $# -gt 0 ]; do
+  case "$1" in
+    -f=* | --frontendBranch=*)
```
[the remark this answers is not in the document]
Comment (nmsger-isento) [reply setup/setup-pib.sh]: Fixed, also for `backend-branch`

File: setup/installation_scripts/local_install.sh [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+  # copy the pib-blockly-server and client from the setup-folder to their targets in the user home-dir
+  cp -r "$PIB_BLOCKLY_SETUP_DIR/pib_blockly_server" "$HOME"
+  cp -r "$PIB_BLOCKLY_SETUP_DIR/pib_blockly_client" "$HOME"
+
+  # build the pib-blockly-server
+  cd "$PIB_BLOCKLY_SERVER_DIR" || { print ERROR "${PIB_BLOCKLY_SERVER_DIR} not found"; return 1; }
+  echo $(ls)
+  echo $(ls pib_blockly_server)
+  echo $(ls src)
```
[the remark this answers is not in the document]
Comment (nmsger-isento) [reply setup/installation_scripts/local_install.sh]: Removed

File: setup/installation_scripts/local_install.sh [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+  cp -r "$PIB_BLOCKLY_SETUP_DIR/pib_blockly_client" "$HOME"
+
+  # build the pib-blockly-server
+  cd "$PIB_BLOCKLY_SERVER_DIR" || { print ERROR "${PIB_BLOCKLY_SERVER_DIR} not found"; return 1; }
+  npm install
+  npm run build
+
+  # build the pib-blockly-server
+  cd "$PIB_BLOCKLY_SERVER_DIR" || { print ERROR "${PIB_BLOCKLY_SERVER_DIR} not found"; return 1; }
+  echo $(ls)
```
Comment (VladislavStavila) [inline setup/installation_scripts/local_install.sh:311]: can this be removed?

File: setup/installation_scripts/docker_install.sh
Comment (VladislavStavila) [inline setup/installation_scripts/docker_install.sh:1]: when we are using local setup systemd services are created for starting our application at boot for the docker setup this is currently not the case, would it make sense to allign these two setup types in this regard in this story?

File: setup/installation_scripts/local_install.sh [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+  pip install -r "$BACKEND_DIR/ros_packages/motors/requirements.txt"
+  pip install "$BACKEND_DIR/ros_packages/motors/pib_motors"
+  print INFO "Installed motors dependencies"
+
+  # move ros-packages into working directory
+  cp -r "$BACKEND_DIR/ros_packages/." "$ROS_WORKING_DIR/src"
+  sudo chmod -R 700 "$ROS_WORKING_DIR"
+
+
+  print INFO "Setting up bootservices"
```
Comment (VladislavStavila) [inline setup/installation_scripts/local_install.sh:189]: bootservices --> boot-services?

File: setup/installation_scripts/docker_install.sh
Comment (nmsger-isento) [reply setup/installation_scripts/docker_install.sh:1]: The docker containers are set to always restart in the `docker-compose.yaml`. This means the docker daemon will start the containers on startup as well. Thanks to that, it is not necessary to use a systemd service to start the docker containers.

File: setup/installation_scripts/local_install.sh (hunk 4 above)
Comment (nmsger-isento) [reply setup/installation_scripts/local_install.sh]: removed

File: setup/installation_scripts/local_install.sh (hunk 5 above)
Comment (nmsger-isento) [reply setup/installation_scripts/local_install.sh]: changed

**Review by VladislavStavila — APPROVED** [review]

**Pull request merged**